engine/lib/group.php

```diff
@@ -69,11 +69,57 @@ function remove_group_tool_option($name) {
 		return;
 	}
 
-	foreach ($CONFIG->group_tool_options as $i => $option) {
-		if ($option->name == $name) {
-			unset($CONFIG->group_tool_options[$i]);
-		}
+	if (!isset($CONFIG->group_tool_options[$name])) {
+		return;
+	}
+	
+	unset($CONFIG->group_tool_options[$name]);
+}
+
+/**
+ * Checks if a group has a specific tool enabled.
+ * Forward to the group if the tool is disabled.
+ *
+ * @param string $option     The group tool option to check
+ * @param int    $group_guid The group that owns the page. If not set, this
+ *                           will be pulled from elgg_get_page_owner_guid().
+ *
+ * @return void
+ * @since 1.9.0
+ */
+function elgg_group_tool_gatekeeper($option, $group_guid = null) {
+	$group_guid = $group_guid ?: elgg_get_page_owner_guid();
+	
+	$group = get_entity($group_guid);
+	if (!$group instanceof \ElggGroup) {
+		return;
+	}
+	
+	if ($group->isToolEnabled($option)) {
+		return;
 	}
+	
+	register_error(elgg_echo('groups:tool_gatekeeper'));
+	forward($group->getURL(), 'group_tool');
+}
+
+/**
+ * Function to return available group tool options
+ *
+ * @param \ElggGroup $group optional group
+ *
+ * @return array
+ */
```

Comment:
@since

also should mentioned the removed function groups_get_group_tool_options